### ISSUE TITLE
Improve docs on building and testing

### ISSUE THREAD

**Issue by iamolegga** [containous/traefik#6891]
### Do you want to request a *feature* or report a *bug*?

Feature

### What did you expect to see?

There is not clean description how to run linting and validation properly before commit and open PR, so I have to open different `make` files and try hardly to understand how to check if everything is ok. And after that I have to understand what libs are required for validation just running again and again and checking errors one by one (if running locally without docker), for example:

```sh
➜  traefik add-headers-to-response ✓ PRE_TARGET= make validate-files
./script/make.sh generate validate-lint validate-misspell
---> Making bundle: generate (in /Users/[user]/go/src/github.com/containous/traefik/script)
autogen/genstatic/gen.go

---> Making bundle: validate-lint (in /Users/[user]/go/src/github.com/containous/traefik/script)

---> Making bundle: validate-misspell (in /Users/[user]/go/src/github.com/containous/traefik/script)
/Users/[user]/go/src/github.com/containous/traefik/script/validate-misspell: line 10: misspell: command not found
make: *** [validate-files] Error 127
```

Running without `PRE_TARGET` shows even worst error, that's really hard to understand, it fails on `golangci-lint` but it's installed:

```sh
➜  traefik add-headers-to-response ✓ make validate-files
docker build  -t "traefik-dev:add-headers-to-response" -f build.Dockerfile .
Sending build context to Docker daemon  152.1MB
Step 1/13 : FROM golang:1.14-alpine
 ---> 3289bf11c284
Step 2/13 : RUN apk --update upgrade     && apk --no-cache --no-progress add git mercurial bash gcc musl-dev curl tar ca-certificates tzdata     && update-ca-certificates     && rm -rf /var/cache/apk/*
 ---> Using cache
 ---> 7826ff6ba5fe
Step 3/13 : ARG DOCKER_VERSION=18.09.7
 ---> Using cache
 ---> b4298a3147a4
Step 4/13 : RUN mkdir -p /usr/local/bin     && curl -fL https://download.docker.com/linux/static/stable/x86_64/docker-${DOCKER_VERSION}.tgz     | tar -xzC /usr/local/bin --transform 's#^.+/##x'
 ---> Using cache
 ---> d7ef9da5f661
Step 5/13 : RUN mkdir -p /usr/local/bin     && curl -fsSL -o /usr/local/bin/go-bindata https://github.com/containous/go-bindata/releases/download/v1.0.0/go-bindata     && chmod +x /usr/local/bin/go-bindata
 ---> Using cache
 ---> c0fd6db37ea4
Step 6/13 : RUN curl -sfL https://install.goreleaser.com/github.com/golangci/golangci-lint.sh | bash -s -- -b $GOPATH/bin v1.26.0
 ---> Using cache
 ---> 42e4f34cd5cd
Step 7/13 : RUN  curl -sfL https://raw.githubusercontent.com/client9/misspell/master/install-misspell.sh | bash -s -- -b $GOPATH/bin v0.3.4
 ---> Using cache
 ---> 57db10ee1eab
Step 8/13 : RUN curl -sfL https://install.goreleaser.com/github.com/goreleaser/goreleaser.sh | sh
 ---> Using cache
 ---> 6692ec14e3a9
Step 9/13 : WORKDIR /go/src/github.com/containous/traefik
 ---> Using cache
 ---> cca99452456f
Step 10/13 : COPY go.mod .
 ---> Using cache
 ---> 06084e66122b
Step 11/13 : COPY go.sum .
 ---> Using cache
 ---> a51436c4c248
Step 12/13 : RUN GO111MODULE=on GOPROXY=https://proxy.golang.org go mod download
 ---> Using cache
 ---> ea7e8763322f
Step 13/13 : COPY . /go/src/github.com/containous/traefik
 ---> d6061731ce26
Successfully built d6061731ce26
Successfully tagged traefik-dev:add-headers-to-response
docker run --add-host=host.docker.internal:127.0.0.1  -e "TEST_CONTAINER=1" -v "/var/run/docker.sock:/var/run/docker.sock"   -e OS_ARCH_ARG -e OS_PLATFORM_ARG -e TESTFLAGS -e VERBOSE -e VERSION -e CODENAME -e TESTDIRS -e CI -e CONTAINER=DOCKER		 -v "/Users/[user]/go/src/github.com/containous/traefik/"dist":/go/src/github.com/containous/traefik/"dist"" "traefik-dev:add-headers-to-response" ./script/make.sh generate validate-lint validate-misspell
---> Making bundle: generate (in /go/src/github.com/containous/traefik/script)
removed 'autogen/genstatic/gen.go'

---> Making bundle: validate-lint (in /go/src/github.com/containous/traefik/script)
/go/src/github.com/containous/traefik/script/validate-lint: line 3:  4084 Killed                  golangci-lint run
make: *** [validate-files] Error 137
➜  traefik add-headers-to-response ✓ golangci-lint --version
golangci-lint has version 1.27.0 built from fb74c2e on 2020-05-13T18:45:55Z
```

To summarise: I think there should be more details on contribution process 🙂By the way, thanks for this awesome product 👍
